Automattic/woocommerce-payments
# Add Forter fraud tracking

Fixes https://github.com/Automattic/woocommerce-payments-server/issues/515

#### Changes proposed in this Pull Request

- If there's a `fraud_services.forter` key in the `GET /account` response, enqueue the Forter JS loader on all WCPay pages.
- That JS will generate a cookie that will need to be passed to the server. So, I added logic to do that.

#### Testing instructions

- Use the `add/forter` branch in the server.
- Set `define( 'WCPAY_FORTER_ENABLED', true )` in the server, and the required credentials (`WCPAY_FORTER_SANDBOX_SITE_ID` and `WCPAY_FORTER_SANDBOX_API_KEY`).
- Flush the account cache in the client or wait 2 hours.
- Navigate to any WCPay admin page, check that the Forter JS script has been enqueued.
- Check to see that now there's a `forterToken` cookie in your browser.
- Navigate to any other admin page.
- Check the WCPay logs, you'll see that a request has been made to the `/tracking/forter-token` WCPay server endpoint.
- On the server, you'll be able to see that a request has been sent to Forter (you can see some logs in `/tmp/logstash.log`, that part is still a work in progress).
- No matter how many more times you navigate around WP-Admin, the request won't be sent again.

Note that this change is completely backwards-compatible, if you use the `trunk` branch in the server everything should work just fine.
